File: python/1-50/50. Pow(x, n).py
class Solution(object):
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        if n == 0:
            return 1
        if n == -1:
            return 1.0/x
        if n  % 2 == 0:
            return self.myPow(x * x, n / 2)
        else:
            return self.myPow(x * x, n / 2) * x

File: cpp/1-50/50. Pow(x, n).cpp
class Solution {
public:
    double myPow(double x, int n) {
        long p = n;
        if(p < 0)
        {
        	x = 1/x;
        	p = -p;
        }
        double res = 1;
        while(p > 0){
        	if(p & 1)
        		res *= x;
        	p  >>= 1;
        	x *= x;
        }
        return res;
    }
};